spring-kafka/src/test/java/org/springframework/kafka/retrytopic/AsyncCompletableFutureRetryTopicClassLevelIntegrationTests.java

```diff
@@ -0,0 +1,966 @@
+
+	private final static String MAIN_TOPIC_CONTAINER_FACTORY = "kafkaListenerContainerFactory";
+
+	@Autowired
+	private KafkaTemplate<String, String> kafkaTemplate;
+
+	@Autowired
+	private CountDownLatchContainer latchContainer;
+
+	@Autowired
+	DestinationTopicContainer topicContainer;
+
+	@Test
+	void shouldRetryFirstTopic(@Autowired KafkaListenerEndpointRegistry registry) {
+		kafkaTemplate.send(FIRST_TOPIC, "Testing topic 1");
+		assertThat(topicContainer.getNextDestinationTopicFor("firstTopicId", FIRST_TOPIC).getDestinationName())
+				.isEqualTo("myRetryTopic1-retry");
+		assertThat(awaitLatch(latchContainer.countDownLatch1)).isTrue();
+		assertThat(awaitLatch(latchContainer.customDltCountdownLatch)).isTrue();
+		assertThat(awaitLatch(latchContainer.customErrorHandlerCountdownLatch)).isTrue();
+		assertThat(awaitLatch(latchContainer.customMessageConverterCountdownLatch)).isTrue();
+		registry.getListenerContainerIds().stream()
+				.filter(id -> id.startsWith("first"))
+				.forEach(id -> {
+					ConcurrentMessageListenerContainer<?, ?> container = (ConcurrentMessageListenerContainer<?, ?>) registry
+							.getListenerContainer(id);
+					if (id.equals("firstTopicId")) {
+						assertThat(container.getConcurrency()).isEqualTo(2);
+					}
+					else {
+						assertThat(container.getConcurrency())
+								.describedAs("Expected %s to have concurrency", id)
+								.isEqualTo(1);
+					}
+				});
+	}
+
+	@Test
+	void shouldRetrySecondTopic() {
+		kafkaTemplate.send(SECOND_TOPIC, "Testing topic 2");
+		assertThat(awaitLatch(latchContainer.countDownLatch2)).isTrue();
+		assertThat(awaitLatch(latchContainer.customDltCountdownLatch)).isTrue();
+	}
+
+	@Test
+	void shouldRetryThirdTopicWithTimeout(
+			@Autowired KafkaAdmin admin,
+			@Autowired KafkaListenerEndpointRegistry registry) throws Exception {
+
+		kafkaTemplate.send(THIRD_TOPIC, "Testing topic 3");
+		assertThat(awaitLatch(latchContainer.countDownLatch3)).isTrue();
+		assertThat(awaitLatch(latchContainer.countDownLatchDltOne)).isTrue();
+		Map<String, TopicDescription> topics = admin.describeTopics(THIRD_TOPIC, THIRD_TOPIC + "-dlt", FOURTH_TOPIC);
+		assertThat(topics.get(THIRD_TOPIC).partitions()).hasSize(2);
+		assertThat(topics.get(THIRD_TOPIC + "-dlt").partitions()).hasSize(3);
+		assertThat(topics.get(FOURTH_TOPIC).partitions()).hasSize(2);
+		AtomicReference<Method> method = new AtomicReference<>();
+		ReflectionUtils.doWithMethods(KafkaAdmin.class, m -> {
+			m.setAccessible(true);
+			method.set(m);
+		}, m -> m.getName().equals("newTopics"));
+		@SuppressWarnings("unchecked")
+		Collection<NewTopic> weededTopics = (Collection<NewTopic>) method.get().invoke(admin);
+		AtomicInteger weeded = new AtomicInteger();
+		weededTopics.forEach(topic -> {
+			if (topic.name().equals(THIRD_TOPIC) || topic.name().equals(FOURTH_TOPIC)) {
+				assertThat(topic).isExactlyInstanceOf(NewTopic.class);
+				weeded.incrementAndGet();
+			}
+		});
+		assertThat(weeded.get()).isEqualTo(2);
```

Comment:
The `AsyncCompletableFutureRetryTopicClassLevelIntegrationTests` demonstrates that the async retry feature works well in `RetryTopicClassLevelIntegrationTests`. It means the synchronous condition. (`ConsumerRecord` succeed or fail one by one)

So, I did copy and paste `RetryTopicClassLevelIntegrationTests` and replace `KafkaListener`'s return type to `Mono<?>` or `CompletableFuture<?>`.

In the `RetryTopicClassLevelIntegrationTests`, `weeded` seems to demonstrate that some new topics should be created. 
<img width="610" alt="image" src="https://github.com/user-attachments/assets/65f67f32-5aff-4f70-8343-5c9580e0b621">

As a result of `admin.newTopics()`, instance `weededTopics` has two types (`TopicForRetryable`, `NewTopic`).
I think that the writer of `RetryTopicClassLevelIntegrationTests` want to verify whether expected `NewTopic`s are created. 

If there are no major issues, I would like to follow the assumptions of `RetryTopicClassLevelIntegrationTests` as well. 
This is because I believe that `AsyncRetry` should also satisfy the `synchronous retry` condition where offsets are processed sequentially, given that AsyncRetry may or may not process offsets in sequence.

Reply:
I still don't see how this `weeded` are relevant to async handling.
That is really not a part of retry topic behavior, but rather its configuration.
So, having same verifications in two different places is redundant.